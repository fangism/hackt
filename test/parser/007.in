// this should be a legitimate PRS definition, should pass

defproc myinverter(node a; node b) {
prs {
	a		-> b-
	~a		-> b+
}
}

