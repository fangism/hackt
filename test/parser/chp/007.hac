// "007.hac" -- should fail, wrong number of expressions

defproc one_shot(chan?(bool) L; chan!(bool) R) {
chp {
	R!(false, true)
}
}

