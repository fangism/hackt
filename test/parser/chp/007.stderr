You passed the wrong number of expressions (2) to a channel-send that expects 1 arguments.
At least one type error in expr-list in [5:4..16]
ERROR: at least one error in CHP body.
